File: src/Core/NetPad.Application/UiInterop/WindowState.cs
namespace NetPad.UiInterop;

public class WindowState
{
    public WindowState(WindowViewStatus viewStatus, bool isAlwaysOnTop)
    {
        ViewStatus = viewStatus;
        IsAlwaysOnTop = isAlwaysOnTop;
    }

    public WindowViewStatus ViewStatus { get; }
    public bool IsAlwaysOnTop { get; }
    public bool IsMinimized => ViewStatus == WindowViewStatus.Minimized;
    public bool IsMaximized => ViewStatus == WindowViewStatus.Maximized;
}

File: src/Apps/NetPad.Apps.App/App/src/core/@application/windows/window-state.ts
export enum WindowViewStatus {
    "Unknown" = "Unknown",
    "Minimized" = "Minimized",
    "UnMaximized" = "UnMaximized",
    "Maximized" = "Maximized",
}

export class WindowState {
    constructor(public viewStatus: WindowViewStatus, public isAlwaysOnTop: boolean) {
    }

    public get isMinimized(): boolean {
        return this.viewStatus === WindowViewStatus.Minimized;
    }

    public get isMaximized(): boolean {
        return this.viewStatus === WindowViewStatus.Maximized;
    }
}
